Code/GraphMol/ChemReactions/catch_tests.cpp

```diff
@@ -1164,6 +1198,11 @@ TEST_CASE("CXSMILES for reactions", "[cxsmiles]") {
     REQUIRE(rxn);
     CHECK(getSubstanceGroups(*rxn->getReactants()[0]).size() == 2);
     CHECK(getSubstanceGroups(*rxn->getProducts()[0]).size() == 2);
+
+    auto roundtrip = v2::ReactionParser::ReactionFromSmarts(ChemicalReactionToRxnCXSmarts(*rxn));
+    REQUIRE(roundtrip);
+    CHECK(getSubstanceGroups(*roundtrip->getReactants()[0]).size() == 2);
+    CHECK(getSubstanceGroups(*roundtrip->getProducts()[0]).size() == 2);
   }
   SECTION("link nodes") {
```

Comment:
What happens if you test the roundtrip CXSmarts for linknodes? Like
```
auto roundtrip = v2::ReactionParser::ReactionFromSmarts(ChemicalReactionToRxnCXSmarts(*rxn));
REQUIRE(roundtrip);
CHECK(roundtrip->getReactants().size() == 2);
CHECK(roundtrip->getProducts().size() == 1);
    CHECK(
        !roundtrip->getReactants()[0]->hasProp(common_properties::molFileLinkNodes));
    std::string lns;
    CHECK(roundtrip->getReactants()[1]->getPropIfPresent(
        common_properties::molFileLinkNodes, lns));
    CHECK(lns == "1 3 2 2 3 2 7");
    CHECK(roundtrip->getProducts()[0]->getPropIfPresent(
        common_properties::molFileLinkNodes, lns));
    CHECK(lns == "2 5 2 5 4 5 7");
```

Reply:
Link nodes are not carried through the round trip.